examples/example.ts

```diff
@@ -0,0 +1,398 @@
+import { BlockFrostAPI } from "@blockfrost/blockfrost-js";
+import invariant from "@minswap/tiny-invariant";
+import {
+  Address,
+  Blockfrost,
+  Constr,
+  Data,
+  Lucid,
+  Network,
+  OutRef,
+  TxComplete,
+  UTxO,
+} from "lucid-cardano";
+
+import {
+  ADA,
+  Asset,
+  BlockfrostAdapter,
+  calculateDeposit,
+  calculateSwapExactIn,
+  calculateSwapExactOut,
+  calculateWithdraw,
+  calculateZapIn,
+  Dex,
+  NetworkId,
+  PoolDatum,
+  PoolState,
+} from "../src";
+
+async function main(): Promise<void> {
+  const network: Network = "Preprod";
+  const networkId: NetworkId = NetworkId.TESTNET;
+  const blockfrostProjectId = "<YOUR_BLOCKFROST_PROJECT_ID>";
+  const blockfrostUrl = "https://cardano-preprod.blockfrost.io/api/v0";
+
+  const address = "<YOUR_ADDRESS>";
+  const lucid = await getBackendLucidInstance(
+    network,
+    blockfrostProjectId,
+    blockfrostUrl,
+    address
+  );
+
+  const blockfrostAdapter = new BlockfrostAdapter({
+    blockFrost: new BlockFrostAPI({
+      projectId: blockfrostProjectId,
+      network: "preprod",
+    }),
+    networkId: networkId,
+  });
+
+  const utxos = await lucid.utxosAt(address);
+
+  const txComplete = await _swapExactInTxExample(
+    network,
+    lucid,
+    blockfrostAdapter,
+    address,
+    utxos
+  );
+  const signedTx = await txComplete
+    .signWithPrivateKey("<YOUR_PRIVATE_KEY>")
+    .complete();
+  const txId = await signedTx.submit();
+  // eslint-disable-next-line no-console
+  console.log(`Transaction submitted successfully: ${txId}`);
+}
+
+async function getPoolById(
+  network: Network,
+  blockfrostAdapter: BlockfrostAdapter,
+  poolId: string
+): Promise<{ poolState: PoolState; poolDatum: PoolDatum }> {
+  const pool = await blockfrostAdapter.getPoolById({
+    id: poolId,
+  });
+  if (!pool) {
+    throw new Error(`Not found PoolState of ID: ${poolId}`);
+  }
+
+  const rawRoolDatum = await blockfrostAdapter.getDatumByDatumHash(
+    pool.datumHash
+  );
+  const poolDatum = PoolDatum.fromPlutusData(
+    network === "Mainnet" ? NetworkId.MAINNET : NetworkId.TESTNET,
+    Data.from(rawRoolDatum) as Constr<Data>
+  );
+  return {
+    poolState: pool,
+    poolDatum: poolDatum,
+  };
+}
```

Comment:
On `preprod` testnet, the total liquidity acquired from this `poolDatum` seems to be incorrect.

For example, when accessing the `ADA/MIN` datum, I get a total liquidity value of `262828735`.

When I use this number for calculating the Zap In expected LP with 50 ADA, I get `237,556`. That seems correct, but when I submit the zap in order I get back `2,143,510` LP, an order of magnitude difference.

When I look at the datum of the order contract, it's telling me a completely different number for the total liquidity: `8,197,014`.

Looking at the datum more closely, it looks like it is operating as a pool with ADA on one side and LP on the other according to the assets in the datum.

It seems to me that the datum is either incorrect for the `ADA/MIN` pool or a different address needs to be used for the `datum`. Or theres a bug in how the order contract is storing datum in the pool UTxO.

Reply:
Heres the completed order transaction for reference:
https://preprod.cardanoscan.io/transaction/75c0646d5879bc3a4318eb4b36582c8b1a6180c2125e12c27fb6cc8d3fc019ce?tab=utxo

Ignore what I have in the datum with an `8` in the minimum receive part. I found a bug in my code that inverted a set of numbers.

Reply:
@theeldermillenial To avoid empty pool, in creating pool transaction, we burned 10 LP Asset of Pool Creator. So the Total Liquidity Pool in the datum is always larger than quantity of LP Asset 10

Reply:
@h2physics That accounts for 10LP. What about the order of magnitude difference in LP?
`262,828,735` is a big difference compared to `8,197,014`.

That seems to be leading to an order of magnitude difference in the estimated zap in deposit.